types/mempool/mempool.go

```diff
@@ -14,10 +14,10 @@ type Mempool interface {
 
 	// Select returns an Iterator over the app-side mempool. If txs are specified,
 	// then they shall be incorporated into the Iterator. The Iterator is not thread-safe to use.
-	Select(context.Context, [][]byte) Iterator
+	Select(context.Context, []sdk.Tx) Iterator
```

Comment:
why the change here?

Reply:
As otherwise in server/v2 we would decode twice if the mempool implementation decodes the txs (ours do not, but we cannot know about users):

https://github.com/cosmos/cosmos-sdk/blob/95b8092/server/v2/cometbft/handlers/defaults.go#L52
https://github.com/cosmos/cosmos-sdk/blob/95b8092/server/v2/cometbft/handlers/defaults.go#L72

Reply:
Should we use a generic instead of either `[]byte` or `sdk.Tx`? if we use `sdk.Tx` how will this work with server/v2?

Reply:
err.. this interface is only used for app v1 correct?

Reply:
Yea, the one of server/v2 is identical but use generic